diff -ruN debian.orig/control debian/control
--- debian.orig/control	2022-08-04 15:19:35.000000000 +0900
+++ debian/control	2024-09-24 04:26:53.703535346 +0900
@@ -4,7 +4,7 @@
 Maintainer: Nobuhiro Iwamatsu <iwamatsu@debian.org>
 Build-Depends: debhelper-compat (= 13),
                libgmp-dev,
-               llvm-dev,
+               llvm-19-dev,
                libmassivethreads-dev
 Standards-Version: 4.5.1
 Rules-Requires-Root: no
@@ -14,7 +14,8 @@
 
 Package: smlsharp
 Architecture: amd64
-Depends: ${shlibs:Depends}, ${misc:Depends}
+Depends: ${shlibs:Depends}, ${misc:Depends},
+         gcc, llvm-19, libc-dev, libgmp-dev, libmassivethreads-dev
 Description: Standard ML compiler with practical extensions
  SML# is a variant of Standard ML programming
  language with several practical extentions,
diff -ruN debian.orig/docs debian/docs
--- debian.orig/docs	2022-08-04 15:19:35.000000000 +0900
+++ debian/docs	2024-09-24 05:16:35.990066540 +0900
@@ -0,0 +1,2 @@
+README.md
+LICENSE
diff -ruN debian.orig/install debian/install
--- debian.orig/install	1970-01-01 09:00:00.000000000 +0900
+++ debian/install	2024-09-24 05:16:10.585940618 +0900
@@ -0,0 +1,10 @@
+src/smlformat/README.txt usr/share/doc/smlsharp/smlformat
+src/smlformat/doc/OVERVIEW_en.txt usr/share/doc/smlsharp/smlformat
+src/smlformat/doc/OVERVIEW_ja.txt usr/share/doc/smlsharp/smlformat
+src/smlformat/doc/PPAlgorithm.txt usr/share/doc/smlsharp/smlformat
+src/smlformat/doc/SimpleTreePP.sml usr/share/doc/smlsharp/smlformat
+src/ml-lex/README usr/share/doc/smlsharp/smllex
+src/ml-lex/README.smlsharp usr/share/doc/smlsharp/smllex
+src/ml-yacc/COPYRIGHT usr/share/doc/smlsharp/smlyacc
+src/ml-yacc/README usr/share/doc/smlsharp/smlyacc
+src/ml-yacc/README.smlsharp usr/share/doc/smlsharp/smlyacc
diff -ruN debian.orig/rules debian/rules
--- debian.orig/rules	2022-08-04 15:19:35.000000000 +0900
+++ debian/rules	2024-09-24 04:26:53.703535346 +0900
@@ -10,9 +10,10 @@
 %:
 	dh $@
 
+override_dh_autoreconf:
+
 override_dh_auto_configure:
-	CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" \
-	./configure --prefix=/usr
+	dh_auto_configure -- --with-llvm=/usr/lib/llvm-19
 
 override_dh_auto_build:
 	dh_auto_build -- stage
diff -ruN debian.orig/smlsharp.manpages debian/smlsharp.manpages
--- debian.orig/smlsharp.manpages	2022-08-04 15:19:35.000000000 +0900
+++ debian/smlsharp.manpages	2024-09-24 05:20:30.250980157 +0900
@@ -1 +1,4 @@
-debian/smlsharp.1
+man/smlformat.1
+man/smllex.1
+man/smlsharp.1
+man/smlyacc.1
